### PR TITLE
Improve: more parens in patterns

### PR DESCRIPTION
**Pull request by hhugo**
Put parens around complex pattern on the lhs of let binding.

### CODE REVIEW

**Comment by jberdine** [comment]
Motivation question: would it make sense to add these parens only if the pattern does not fit on one line, or you think its better even for the one-liners?

**Comment by hhugo** [comment]
I personally like it better even for one-liners.
One of the motivation is to make the difference with function declaration more apparent
```
let f x = e
let `f x = e
let F x = e
```

**Comment by jberdine** [comment]
Ok, I see. The point about function definitions never occurred to me because colors:
image

File: src/Ast.ml [hunk 1]
```diff
@@ -749,6 +749,14 @@ end = struct
     | ( Pat {ppat_desc= Ppat_construct _}
       , Ppat_construct ({txt= Lident "::"}, _) ) ->
         true
+    | ( (Exp {pexp_desc= Pexp_let _; _} | Str {pstr_desc= Pstr_value _; _})
+      , ( Ppat_construct (_, Some _)
+        | Ppat_variant (_, Some _)
+        | Ppat_or _ | Ppat_alias _ ) ) ->
+        (* let (Some p) = e *)
+        (* let (P1 | P2) = e *)
+        (* let (pat as ident) = e *)
```
Comment (jberdine) [inline src/Ast.ml]: Nit: Do you think that it is valuable to have only a few such comments indicating the concrete syntax corresponding to the parsetree constructors? I was basically assuming that anyone reading to code would either be somewhat familiar with the parsetree, or willing to read it as they went.

Comment (hhugo) [reply src/Ast.ml]: It was only meaningful internally to help understand what diffs can be upstream. Now removed.

**Comment by jberdine** [comment]
Ok, I'm not really convinced about the cases except for Ppat_alias, but it's not very invasive so lets see if there are complaints and then we can see-saw back, add an option, etc.

**Pull request merged**